### ISSUE TITLE
Drawing image with src param does not work correctly

### ISSUE THREAD

**Issue by InAnYan** [ggez/ggez#1267]
**Describe the bug**
I was trying to draw a fraction of an image by setting the `src` parameter of `DrawParam`.
From the documentation, it accepts float numbers from 0 to 1. I was trying to draw a half of an image, but I got strange output.

Maybe I don't understand how `src` works? Maybe I've passed wrong numbers?

**To Reproduce**
Here is the code (`main.rs`):
```rust
use ggez::{event::{EventHandler, self}, ContextBuilder, graphics::{Image, Canvas, Color, DrawParam, Rect}, glam::Vec2};

struct Game {
    image: Image
}

impl EventHandler for Game {
    fn update(&mut self, _ctx: &mut ggez::Context) -> Result<(), ggez::GameError> {
        Ok(()) 
    }

    fn draw(&mut self, ctx: &mut ggez::Context) -> Result<(), ggez::GameError> {
        let mut canvas = Canvas::from_frame(ctx, Color::BLACK);

        let param = DrawParam::new()
            .src(Rect::new(0.5, 0.0, 1.0, 1.0))
            .dest(Vec2::new(100.0, 100.0));

        canvas.draw(&self.image, param);

        canvas.finish(ctx)
    }
}

fn main() {
    let (ctx, event_loop) = ContextBuilder::new("title", "author")
        .build()
        .expect("could not create ggez context");

    let image = Image::from_path(&ctx, "/cat.jpg")
        .expect("could not load cat image");

    let game = Game {
        image,
    };

    event::run(ctx, event_loop, game);
}
```

I put an image (`cat.jpg`) to the `target/debug/resources/`:
![cat](https://github.com/ggez/ggez/assets/13097618/26e8861a-86bb-4856-95f2-5064e82e8a63)

**Expected behavior**
I thought that at the point 100x100 px there would be a half of the cat image, but there is also some stretching thing.

**Screenshots or pasted code**
![image](https://github.com/ggez/ggez/assets/13097618/e19edc25-433e-45ed-9ca4-aa483a3aed84)

**Hardware and Software:**
 - ggez version: release 0.9.3
 - OS: Windows 11
 - Graphics card: NVidia GeForce GTX 1650
 - Graphics card drivers: game ready, up to date (546.33)

**Comment by jazzfool** [comment]
Rect::new takes x,y,w,h. So with src, when x = 0.5 and w = 1, the src rect extends past the image, with x ranging [0.5, 1.5]. That stretching effect is what happens when you sample an image outside its texel coordinate boundaries (i.e., src coord outside [0, 1]). That effect can be controlled by the "addressing mode" (see https://docs.rs/ggez/latest/ggez/graphics/enum.ClampMode.html), which by default is set to clamp (i.e., the texel coordinates are clamped to nearest texel at the edge).

tldr if you want to sample the right half of the image, set src = Rect::new(0.5, 0.0, 0.5, 1.0), or use Image::uv_rect to do it in pixel coordinates

**Comment by InAnYan** [comment]
Oh, width and height! What a silly mistake. Thank you for the help!

Yes, after the change to numbers the image is correct. The issue is closed.
